### PR TITLE
Add Tests for JsonAdaptedInternship

### PR DESCRIPTION
**Pull request by helloImHai**
Other kids can copy! @duongphammmm hahaha

### CODE REVIEW

File: src/test/java/seedu/address/storage/JsonAdaptedInternshipTest.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
+package seedu.address.storage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static seedu.address.commons.util.JsonUtil.readJsonFile;
```
Comment (wardetu) [inline src/test/java/seedu/address/storage/JsonAdaptedInternshipTest.java]: Only test this in `JsonAddressBookStorageTest`

File: src/test/java/seedu/address/storage/JsonAdaptedInternshipTest.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        Optional<JsonAdaptedInternship> jsonAdaptedInternship =
+                readJsonFile(addToTestDataPathIfNotNull("validInternship.json"),
+                        JsonAdaptedInternship.class);
+        JsonAdaptedInternship expectedJsonAdapatedInternship =
+                new JsonAdaptedInternship(new InternshipBuilder(TypicalInternship.PAYPAL).build());
+        assertEquals(expectedJsonAdapatedInternship, jsonAdaptedInternship.get());
+    }
+
+    @Test
+    public void loadToModel_jsonToInternship_success() throws DataConversionException, IllegalValueException {
```
Comment (wardetu) [inline src/test/java/seedu/address/storage/JsonAdaptedInternshipTest.java]: `toModelType()` converts a `JsonAdaptedInternship` to an `Internship`.

I think we should not include reading of json in here.

**Review by wardetu — CHANGES_REQUESTED** [review]
There are quite several issues that need to be addressed :(

File: src/main/java/seedu/address/storage/JsonAdaptedInternship.java [hunk 3]
```diff
@@ -125,4 +125,17 @@ public Internship toModelType() throws IllegalValueException {
 
         return new Internship(modelName, role, modelFrom, modelTo, description, Set.copyOf(tags), modelId);
     }
+
+    @Override
```
Comment (wardetu) [inline src/main/java/seedu/address/storage/JsonAdaptedInternship.java:142]: I'm on the fence about having this.
Let me think about it.

File: src/test/java/seedu/address/storage/JsonAdaptedInternshipTest.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+                jsonAdaptedInternship.get()::toModelType);
+    }
+
+    @Test
+    public void loadToModel_invalidFromTo_failure() throws DataConversionException {
+        Optional<JsonAdaptedInternship> jsonAdaptedInternship =
+                readJsonFile(addToTestDataPathIfNotNull("invalidFromToInternship.json"),
+                        JsonAdaptedInternship.class);
+        assertThrows(IllegalValueException.class,
+                "The \"to\" field must not precede the \"from\" field.",
```
Comment (duongphammmm) [inline src/test/java/seedu/address/storage/JsonAdaptedInternshipTest.java:271]: might want to change this to be consistent with the error shown when adding item? It was 'from' cannot be later than 'to' field.

**Review by chrisjwelly — APPROVED** [review]
LGTM! :tada:

**Review by wardetu — CHANGES_REQUESTED** [review]
Many test methods are still called `loadToModel` even though they test `toModelType`.
I think the names should change to accurately match.

**Review by wardetu — APPROVED** [review]
LGTM!

**Pull request merged**